extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ContainerAdapter.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.dekorate.deps.kubernetes.api.model.Container;
+import io.dekorate.deps.kubernetes.api.model.ContainerBuilder;
+import io.dekorate.kubernetes.config.Env;
+import io.dekorate.kubernetes.config.Mount;
+import io.dekorate.kubernetes.config.Port;
+import io.dekorate.kubernetes.decorator.AddEnvVarDecorator;
+import io.dekorate.kubernetes.decorator.AddLivenessProbeDecorator;
+import io.dekorate.kubernetes.decorator.AddMountDecorator;
+import io.dekorate.kubernetes.decorator.AddReadinessProbeDecorator;
+import io.dekorate.kubernetes.decorator.ApplyImagePullPolicyDecorator;
+import io.dekorate.utils.Images;
+import io.dekorate.utils.Strings;
+
+/**
+ * Copied from dekorate in order to fix some issues
+ */
+public class ContainerAdapter {
+
+    public static Container adapt(io.dekorate.kubernetes.config.Container container) {
+        String name = container.getName();
+        if (Strings.isNullOrEmpty(name)) {
+            name = Images.getName(container.getImage());
+        }
+
+        ContainerBuilder builder = new ContainerBuilder()
+                .withName(name)
+                .withImage(container.getImage())
+                // this was changed to add working dir
```

Comment:
We will have to apply that upstream as well

Reply:
Yes of course. I just wanted to get these changes in Quarkus so we can include the fixes in 1.6 which is right around the corner.